geomstats/geometry/pullback_metric.py

```diff
@@ -221,38 +139,12 @@ def second_fundamental_form(self, base_point):
             Second fundamental form :math:`\RN{2}(p)_{ij}^\alpha` where the
              :math:`\alpha` index is first.
         """
-        initial_ndim = base_point.ndim
-        base_point = gs.to_ndarray(base_point, to_ndim=2)
-        second_fundamental_forms = []
-        for point in base_point:
-            christoffels = self.christoffels(point)
-
-            jacobian_ai = self.jacobian_immersion(point)
-            if self.dim == 1 and jacobian_ai.ndim > 2:
-                jacobian_ai = gs.squeeze(jacobian_ai, axis=-1)
-
-            hessian_aij = self.hessian_immersion(point)
-            if self.dim == 1 and hessian_aij.ndim > 3:
-                hessian_aij = gs.squeeze(hessian_aij, axis=-1)
-
-            second_fundamental_form_aij = []
-            for a in range(self.embedding_metric.dim):
-                jacobian_a = jacobian_ai[a]
-                hessian_a = hessian_aij[a]
-
-                second_fundamental_form_a = hessian_a - gs.einsum(
-                    "kij,k->ij", christoffels, jacobian_a
-                )
-                second_fundamental_form_aij.append(second_fundamental_form_a)
-
-            second_fundamental_forms_aij = gs.stack(second_fundamental_form_aij, axis=0)
-            second_fundamental_forms.append(second_fundamental_forms_aij)
-        second_fundamental_forms = gs.stack(second_fundamental_forms, axis=0)
-        return (
-            second_fundamental_forms[0]
-            if initial_ndim == 1
-            else second_fundamental_forms
-        )
+        christoffels = self.christoffels(base_point)
+
+        jacobian = self._space.jacobian_immersion(base_point)
+        hessian = self._space.hessian_immersion(base_point)
+
+        return hessian - gs.einsum("...kij,...dk->...dij", christoffels, jacobian)
```

Comment:
good if k denotes the index of the derivative \partial_k

Reply:
I haven't changed the expression, only added `...` for multiple point.